Gems/Multiplayer/Code/Source/Debug/MultiplayerDebugMultiplayerMetrics.cpp

```diff
@@ -0,0 +1,222 @@
+    {
+        const MultiplayerStats::Metric propertyUpdatesSent = stats.CalculateComponentPropertyUpdateSentMetrics(netComponentId);
+        const MultiplayerStats::Metric propertyUpdatesRecv = stats.CalculateComponentPropertyUpdateRecvMetrics(netComponentId);
+        const MultiplayerStats::Metric rpcsSent = stats.CalculateComponentRpcsSentMetrics(netComponentId);
+        const MultiplayerStats::Metric rpcsRecv = stats.CalculateComponentRpcsRecvMetrics(netComponentId);
+        const uint64_t totalCalls = propertyUpdatesSent.m_totalCalls + propertyUpdatesRecv.m_totalCalls + rpcsSent.m_totalCalls + rpcsRecv.m_totalCalls;
+        const uint64_t totalBytes = propertyUpdatesSent.m_totalBytes + propertyUpdatesRecv.m_totalBytes + rpcsSent.m_totalBytes + rpcsRecv.m_totalBytes;
+        float callsPerSecond = 0.0f;
+        float bytesPerSecond = 0.0f;
+        AccumulatePerSecondValues(stats, propertyUpdatesSent, callsPerSecond, bytesPerSecond);
+        AccumulatePerSecondValues(stats, propertyUpdatesRecv, callsPerSecond, bytesPerSecond);
+        AccumulatePerSecondValues(stats, rpcsSent, callsPerSecond, bytesPerSecond);
+        AccumulatePerSecondValues(stats, rpcsRecv, callsPerSecond, bytesPerSecond);
+        return DrawMetricsRow(name, true, totalCalls, totalBytes, callsPerSecond, bytesPerSecond);
+    }
+
+    void DrawComponentDetails(const MultiplayerStats& stats, NetComponentId netComponentId)
+    {
+        MultiplayerComponentRegistry* componentRegistry = GetMultiplayerComponentRegistry();
+        {
+            const MultiplayerStats::Metric metric = stats.CalculateComponentPropertyUpdateSentMetrics(netComponentId);
+            float callsPerSecond = 0.0f;
+            float bytesPerSecond = 0.0f;
+            AccumulatePerSecondValues(stats, metric, callsPerSecond, bytesPerSecond);
+            if (DrawMetricsRow("PropertyUpdates Sent", true, metric.m_totalCalls, metric.m_totalBytes, callsPerSecond, bytesPerSecond))
+            {
+                const MultiplayerStats::ComponentStats& componentStats = stats.m_componentStats[aznumeric_cast<AZStd::size_t>(netComponentId)];
+                for (AZStd::size_t index = 0; index < componentStats.m_propertyUpdatesSent.size(); ++index)
+                {
+                    const PropertyIndex propertyIndex = aznumeric_cast<PropertyIndex>(index);
+                    const char* propertyName = componentRegistry->GetComponentPropertyName(netComponentId, propertyIndex);
+                    const MultiplayerStats::Metric& subMetric = componentStats.m_propertyUpdatesSent[index];
+                    callsPerSecond = 0.0f;
+                    bytesPerSecond = 0.0f;
+                    AccumulatePerSecondValues(stats, subMetric, callsPerSecond, bytesPerSecond);
+                    DrawMetricsRow(propertyName, false, subMetric.m_totalCalls, subMetric.m_totalBytes, callsPerSecond, bytesPerSecond);
+                }
+                ImGui::TreePop();
+            }
+        }
+        {
```

Comment:
nit: Maybe add comment to each functional block to improve code readability ie

// Calculate and display metrics about recieved data
{
}

// Calculate and display RPC sent metrics